consensus/state.go

```diff
@@ -674,9 +674,28 @@ func (cs *State) updateToState(state sm.State) {
 		// to be gathered for the first block.
 		// And alternative solution that relies on clocks:
 		// cs.StartTime = state.LastBlockTime.Add(timeoutCommit)
-		cs.StartTime = cs.config.Commit(cmttime.Now())
+
+		if state.LastBlockHeight == 0 {
+			// state represents the genesis state, hence the StartTime should be set based on the state's TimeoutCommit
+			// we don't use cs.state.TimeoutCommit because that is zero
+			cs.StartTime = cs.config.CommitWithCustomTimeout(cmttime.Now(), state.TimeoutCommit)
+		} else {
+			// if state does not represent the genesis state,
+			// we use the cs.state.TimeoutCommit
+			cs.StartTime = cs.config.CommitWithCustomTimeout(cmttime.Now(), cs.state.TimeoutCommit)
+		}
```

Comment:
[nit] instead of code comments that distract from the functionality, we can move the comments into code. For example

```suggestion
		if state.IsGenesis() {
			// Don't use cs.state.TimeoutCommit because that is zero
			cs.StartTime = cs.config.CommitWithCustomTimeout(cmttime.Now(), state.TimeoutCommit)
		} else {
			cs.StartTime = cs.config.CommitWithCustomTimeout(cmttime.Now(), cs.state.TimeoutCommit)
		}
```

with the addition of

```
func (state State) IsGenesis() bool {
	return state.LastBlockHeight == 0
}

```

Reply:
This comment wasn't addressed.